experimental/lib/Support/BooleanLogic/Lexer.cpp

```diff
@@ -0,0 +1,112 @@
+//===- Lexer.cpp - Lexer for boolean expressions ----------------*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the LexicalAnalyzer class for tokenizing a string
+// representing a boolean logic expression. The constructor processes the input
+// string, identifying tokens and storing them in a list. The GetToken method
+// retrieves the next token, while peek allows looking ahead in the token
+// stream.
+//
+//===----------------------------------------------------------------------===//
+
+#include "experimental/Support/BooleanLogic/Lexer.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/Support/raw_ostream.h"
+#include <string>
+#include <vector>
+
+using namespace llvm;
+using namespace mlir;
+using namespace dynamatic::experimental::boolean;
+
+// Function for reporting syntax errors encountered during lexical analysis.
+void dynamatic::experimental::boolean::syntaxError(char current) {
+  llvm::errs() << "Syntax error in expression! Invalid character " << current
+               << '\n';
+}
+
+LexicalAnalyzer::LexicalAnalyzer(StringRef exp) : expression(exp) {}
+
+// The function tokenize processes a string expression, skipping whitespace and
+// identifying variable tokens from letter-digit sequences. It creates tokens
+// for operators and parentheses, storing all tokens in tokenList. An end token
+// is added to mark the completion of tokenization.
+mlir::LogicalResult LexicalAnalyzer::tokenize() {
+  size_t position = 0;
+  while (position < expression.length()) {
+    char current = expression[position];
+    if (isspace(current)) {
+      position++;
+      continue;
+    }
+    if (isalpha(current)) { // If the current character is a letter, it

```

Comment:
Put the comment aboce the if or in the body, right not it breaks the line and creates weird indexing.